graph/pull.go

```diff
@@ -102,6 +116,93 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 	s.eventsService.Log("pull", logName, "")
 
 	return nil
+
+}
+
+func makeMirrorRepoInfo(repoInfo *registry.RepositoryInfo, mirror string) *registry.RepositoryInfo {
+	mirrorRepo := &registry.RepositoryInfo{
+		RemoteName:    repoInfo.RemoteName,
+		LocalName:     repoInfo.LocalName,
+		CanonicalName: repoInfo.CanonicalName,
+		Official:      false,
+
+		Index: &registry.IndexInfo{
+			Official: false,
+			Secure:   repoInfo.Index.Secure,
+			Name:     mirror,
+			Mirrors:  []string{},
+		},
+	}
+	return mirrorRepo
+}
+
+func configureV2Mirror(repoInfo *registry.RepositoryInfo, s *registry.Service) (*registry.Endpoint, *registry.RepositoryInfo, error) {
+	mirrors := repoInfo.Index.Mirrors
+	if len(mirrors) == 0 && !repoInfo.Index.Official {
```

Comment:
@RichardScothern this is weird, you only call `configureV2Mirror` if `repoInfo.Index.Official` is true, so this block can be ommitted. Cf https://github.com/docker/docker/pull/13374/files#diff-7cbe4277b9a08ded689d34eaae37655eR64